store/src/java/com/zimbra/cs/service/mail/ToXML.java

```diff
@@ -2967,6 +3024,143 @@ private static void addContent(Element elt, MPartInfo mpi, int maxSize, boolean
         }
         // TODO: CDATA worth the effort?
     }
+    /*
+    * Adds Warning Content to html content of Response
+    * */
+    private static void addContentWithExternalWarning(Element elt, MPartInfo mpi, int maxSize, boolean neuter, String defaultCharset, MsgContent wantContent,String warningMsg)
```

Comment:
Why have we created a new duplicate method addContentWithExternalWarning() instead of adding some code block in the existing method addContent()? We can add the below code in addContent() to achieve the same.

`if (!data.contains("<html>")) {
                String bodyContent = data;
                String htmBodyWithWarning = "<html><body><div><p>"+warningMsg+"</p><hr /></div><div style=\"font-size: 12pt; font-family: arial, helvetica, sans-serif; color: rgb(0, 0, 0);\"><div>&nbsp;</div><div>"+bodyContent+"</div>\n";
data = htmBodyWithWarning;
}`

Reply:
To avoid redundancy, added one more parameter to addContent with a logic to add warning and removed addContentWithExternalWarning.
Made necessary changes to the methods calling addContent